api/src/main/java/org/apache/iceberg/NullOrder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Locale;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+public enum NullOrder {
+  NULLS_FIRST("first"), NULLS_LAST("last");
+
+  private final String jsonValue;
+
+  NullOrder(String jsonValue) {
+    this.jsonValue = jsonValue;
+  }
+
+  public static NullOrder fromJsonValue(String jsonValue) {
```

Comment:
What about `toString`/`fromString`? Instead of using `jsonValue`? Since it is just a simple string, there is no need to call out that it is intended for JSON.

Reply:
I see `toString` is used for the SQL-ish representation. Maybe we should just embed this in a parser class instead of having them here.

Reply:
Will move this logic to the parser.

Reply:
Done.